common/src/HttpMessageHelper.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+#if NET451
+using System.Net.Http.Formatting;
+#endif
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    internal static class HttpMessageHelper
+    {
+#if !NET451
```

Comment:
#if NET451
        private static readonly JsonMediaTypeFormatter s_jsonFormatter = new JsonMediaTypeFormatter();
#else
        private const string ApplicationJson = "application/json";
#endif

Reply:
If we want to keep the order of const and static, we can switch the if-else. It just seemed too much to process when it was #if NET451 and then right after #if !NET451

Reply:
sure, I can switch it

Reply:
the way that you have written reads better to me (positive case then negative case); we can probably switch the order of const vs readonly for the sake of readability for this one instance.
@drwill-ms thought?

Reply:
agreed, positive case reads better to me